### DISCUSSION TITLE
3D Terrain rendering

### DISCUSSION THREAD

**Discussion by Tanonin** [osmandapp/OsmAnd#18551] [Maps Related]
When using the new "3D Relief" feature the map does not load to the finest level of detail if you are in altitude. For example at 3000m the highest resolution of the map is 200m scale and and at 500m elevation the more you can zoom is to 50m scale. After that if you continue zooming you start seeing pixelated edges like a raster map. If you zoom furthemore you go "through" the terrain and it's blank.
Can you do something to fix that or the 3D map is supposed to be used at a higher scale so it doesn't matter much? Anyway you need to advise the user that it's gonna happened when using 3D because it really feels like the map is messed up and one don't know that it's dependent of the 3D setting in the first place.
Thanks

**Comment by DmitryAlexei** [comment]
What version of the app are you using? (go to menu > help > about OsmAnd)
you can try beta, some issues with 3d have been resolved

**Reply by Tanonin** [reply]
I'm running 4.5.10. It's good for me for now I just de-activated 3D rendering, I'll wait for the next release then.

**Comment by jrzelazny** [comment]
Looks like a few of the issues have been resolved, but it's still showing a distorted map view when using the 3D terrain.

![Screenshot_20231225-181106](https://github.com/osmandapp/OsmAnd/assets/146493055/19ae7048-c15e-4c17-b6a1-882ec4a3dbe0)
![Screenshot_20231225-131318](https://github.com/osmandapp/OsmAnd/assets/146493055/730d1aa2-8a28-4bea-8fba-261ef208f80d)
